### ISSUE TITLE
Allow multi-line `when` statement following an `else` without requiring braces

### ISSUE THREAD

**Issue by milchopenchev** [detekt/detekt#3708]
This is related to the `MandatoryBracesIfStatements` rule.

## Expected Behavior
To allow this code snippet without lint issues:
```kotlin
fun getList(quality: QualityEnum, specialFlag: Boolean) =
  if (specialFlag) {
    // ...
  } else when (quality) {
    // exhaustive enum when statement
  }
```

## Current Behavior
For the above snipped, there are lint errors thrown that require it to be written in a less legible manner (imho), like so:
```kotlin
fun getList(quality: QualityEnum, specialFlag: Boolean) =
  if (specialFlag) {
    // ...
  } else {
    when (quality) {
      // exhaustive enum when statement
    }
  }
```

## Context
- To check a flag which requires custom behavior or otherwise default to a `when(..)` statement's result
- This can be combined into a single big `when` statement, however, in doing so the exhaustive `when(quality)` is lost, meaning that we have to add an `else` branch, where it would not be necessary otherwise
- The check for `specialFlag` is necessary to be done first (and thus cannot be in an `else` branch of a larger `when` statement, as even when its true, `quality` may have a valid value, which would mean it would incorrectly use another one of the cases instead.

**Comment by cortinico** [comment]
I'm curious to hear other's opinion on this. 

I personally haven't seen this syntax that often: 

```
  } else when (quality) {
```

It definitely looks more readable than instead of having another level of nesting.
I can't seem to find any negative point against allowing this (or at least having an `allowSingleWhenStatement` config flag or so).

Are there any reference to this in the official Kotlin styleguide or so?

**Comment by BraisGabin** [comment]
It's a strange case but if we accept `else if` we should accept `else when`. And I don't thing that we should add any new flag.
